packages/authentication/src/__tests__/fixtures/services/basic-auth-user-service.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/authentication
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {inject} from '@loopback/context';
+import {HttpErrors} from '@loopback/rest';
+import {UserService} from '../../../services/user.service';
+import {UserProfile} from '../../../types';
+import {USER_REPO} from '../keys';
+import {BasicAuthenticationStrategyCredentials} from '../strategies/basic-strategy';
+import {User} from '../users/user';
+import {UserRepository} from '../users/user.repository';
+
+export class BasicAuthenticationUserService
+  implements UserService<User, BasicAuthenticationStrategyCredentials> {
+  constructor(
+    @inject(USER_REPO)
+    private userRepository: UserRepository,
+  ) {}
+
+  async verifyCredentials(
+    credentials: BasicAuthenticationStrategyCredentials,
+  ): Promise<User> {
+    if (!credentials) {
+      throw new HttpErrors.Unauthorized(`'credentials' is null`);
```

Comment:
@jannyHou , we discussed that the strategy resolver would return a JS error object since, in the future, we may use it to resolve strategies for authenticating protocol requests other than http rest requests, and that the strategy resolver that ships in the main code of the authentication package shouldn't be returning any http specific errors. 

However my mock implementations of token service and user service and strategies don't ship in the main code of the authentication package and only exist in the src/__tests__ folder, and so I have chosen to have then throw http errors since I am testing them with a rest server.

Reply:
@emonddr Fair enough 👍